src/main/java/net/syscon/elite/repository/jpa/model/OffenderLanguage.java

```diff
@@ -0,0 +1,68 @@
+package net.syscon.elite.repository.jpa.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.hibernate.annotations.JoinColumnOrFormula;
+import org.hibernate.annotations.JoinColumnsOrFormulas;
+import org.hibernate.annotations.JoinFormula;
+import org.hibernate.annotations.NotFound;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.io.Serializable;
+
+import static org.hibernate.annotations.NotFoundAction.IGNORE;
+
+
+@Entity
+@Table(name = "OFFENDER_LANGUAGES")
+@Getter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Data
+@IdClass(OffenderLanguage.PK.class)
+public class OffenderLanguage {
+
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class PK implements Serializable {
+        @Column(insertable = false, updatable = false)
+        private Long offenderBookId;
+
+        @Column(name = "LANGUAGE_TYPE")
+        private String type;
+
+        @Column(insertable = false, updatable = false, name = "LANGUAGE_CODE")
+        private String code;
+    }
+
+    @Id
+    private Long offenderBookId;
+
+    @Id
+    private String type;
+
+    @Id
+    private String code;
+
+    private String readSkill;
```

Comment:
So kinda on the fence how to represent nullable fields on entities.  So one thing I have done is override getters and return optional's.   I accept this not ideal but just a bit cleaner when dealing with in other places.   If only this were 'Kotlin'....